packages/std/src/serde.rs

```diff
@@ -8,26 +8,75 @@ use serde::{de::DeserializeOwned, Serialize};
 use crate::binary::Binary;
 use crate::errors::{StdError, StdResult};
 
+#[deprecated = "use from_json_slice instead"]
 pub fn from_slice<T: DeserializeOwned>(value: &[u8]) -> StdResult<T> {
-    serde_json_wasm::from_slice(value).map_err(|e| StdError::parse_err(type_name::<T>(), e))
+    from_json_slice(value)
 }
 
+#[deprecated = "use from_json_binary instead"]
 pub fn from_binary<T: DeserializeOwned>(value: &Binary) -> StdResult<T> {
-    from_slice(value.as_slice())
+    from_json_binary(value)
+}
+
+/// Deserializes the given JSON byte vector to a data structure.
+///
+/// Errors if the input is not valid JSON or cannot be deserialized to the given type.
+pub fn from_json_slice<T: DeserializeOwned>(value: &[u8]) -> StdResult<T> {
+    serde_json_wasm::from_slice(value).map_err(|e| StdError::parse_err(type_name::<T>(), e))
+}
+
+/// Deserializes the given JSON bytes to a data structure.
+///
+/// Errors if the input is not valid JSON or cannot be deserialized to the given type.
+pub fn from_json_str<T: DeserializeOwned>(value: &str) -> StdResult<T> {
+    serde_json_wasm::from_str(value).map_err(|e| StdError::parse_err(type_name::<T>(), e))
+}
+
+/// Deserializes the given JSON bytes to a data structure.
+///
+/// Errors if the input is not valid JSON or cannot be deserialized to the given type.
+pub fn from_json_binary<T: DeserializeOwned>(value: &Binary) -> StdResult<T> {
+    from_json_slice(value.as_slice())
 }
```

Comment:
Can we merge binary and slice or even all 3 with and `impl AsRef<[u8]>` argument like [hex::decode](https://docs.rs/hex/latest/hex/fn.decode.html) does?

Reply:
Yes, great idea!